File: solutions/day_04/advent4.2.rb
#!/usr/bin/env ruby

require_relative "./advent4.1"

def calculate_valid_passphrases
  input.map{|i| i.map{|j| j.chars.sort.to_s }}.select{|i| i.uniq.count == i.count}.count
end

if __FILE__ == $0
  p calculate_valid_passphrases
end

File: solutions/day_04/advent4.2.exs
#!/usr/bin/env elixir

IO.read(:stdio, :all)
|> String.split("\n")
|> Enum.map(&(&1 |> String.split))
|> Enum.map(fn (line) ->
  Enum.map(line, fn (word) ->
    word
    |> to_charlist
    |> Enum.sort
    |> List.to_string
  end)
end)
|> Enum.filter(fn(line) ->
  line
  |> Enum.uniq
  |> length == line |> length
end)
|> length
|> IO.puts
